api/src/main/kotlin/io/spine/protodata/renderer/InsertionPoint.kt

```diff
@@ -49,7 +81,93 @@ public interface InsertionPoint {
      * @return the line number at which the insertion point should be added.
      * @see LineNumber
      */
-    public fun locate(lines: List<String>): LineNumber
+    @Deprecated("Use locate(Text) instead.")
+    public fun locate(lines: List<String>): LineNumber {
+        val coords = locate(text(lines))
+        return when (coords.kindCase) {
+            WHOLE_LINE -> LineNumber.at(coords.wholeLine)
+            INLINE -> {
+                logUnsupportedKind()
+                LineNumber.at(coords.inline.cursor.line)
+            }
+            END_OF_FILE -> LineNumber.endOfFile()
+            NOT_IN_FILE -> LineNumber.notInFile()
+            else -> error("Unexpected text coordinates `$coords`.")
+        }
+    }
+
+    private fun logUnsupportedKind() =
+        loggerFor(InsertionPoint::class.java)
+            .atWarning()
+            .withStackTrace(FULL)
+            .log(
+                "`locate(List<String>)` does not support inline insertion. " +
+                        "A whole line insertion will be generated."
+            )
+
+    /**
+     * Locates the site where the insertion point should be added.
+     *
+     * An insertion point should only appear once in a file.
```

Comment:
Taking into account our last conversation, this sentence is not too clear. Can we create an insertion point for, say, a repeating fragment of code picking it via a regular expression? Such an insertion point may serve to suppress common warnings.

In any case, I think this feature requires better documentation.